control-plane-operator/controllers/hostedcontrolplane/konnectivity/reconcile.go

```diff
@@ -39,6 +57,228 @@ const (
 	KubeconfigKey = "kubeconfig"
 )
 
+func ReconcileServerDeployment(deployment *appsv1.Deployment, ownerRef config.OwnerRef, deploymentConfig config.DeploymentConfig, image string) error {
+	ownerRef.ApplyTo(deployment)
+	deployment.Spec = appsv1.DeploymentSpec{
+		Selector: &metav1.LabelSelector{
+			MatchLabels: konnectivityServerLabels(),
+		},
+		Template: corev1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: konnectivityServerLabels(),
+			},
+			Spec: corev1.PodSpec{
+				AutomountServiceAccountToken: pointer.Bool(false),
+				Containers: []corev1.Container{
+					util.BuildContainer(konnectivityServerContainer(), buildKonnectivityServerContainer(image)),
+				},
+				Volumes: []corev1.Volume{
+					util.BuildVolume(konnectivityVolumeServerCerts(), buildKonnectivityVolumeServerCerts),
+					util.BuildVolume(konnectivityVolumeClusterCerts(), buildKonnectivityVolumeClusterCerts),
+					util.BuildVolume(konnectivitySignerCA(), buildKonnectivitySignerCAkonnectivitySignerCAVolume),
+				},
+			},
+		},
+	}
+	deploymentConfig.ApplyTo(deployment)
+	return nil
+}
+
+func konnectivityServerContainer() *corev1.Container {
+	return &corev1.Container{
+		Name: "konnectivity-server",
+	}
+}
+
+const (
```

Comment:
nit: should these be moved to the top of the file?

Reply:
thx, moved it up